packages/vitest/src/integrations/jest-mock.ts

```diff
@@ -185,11 +185,12 @@ function enhanceSpy<TArgs extends any[], TReturns>(
 
   stub.mockClear = () => {
     stub.reset()
+    instances.length = []
```

Comment:
length can't be an array
lets just redeclare it like we do with `onceImplementations`

Reply:
Yeap, This is a stupid mistake on my part 😅

Reply:
Do you really need to re-declare an array of instances? I assumed that it would be useful to empty the array but not destroy all references to it

Reply:
Why do you need the references after it was cleared tho? I would expect this to return undefined: `mock.instances[0]` after clearing.

Reply:
You can always store it before emptying.

Reply:
I was thinking about a scenario where a user saves a reference to an array of instances.
```ts
const {instances} = fn.mock;
```
In this case, overriding will clear the array into mock, but will not change the saved reference:
```ts
fn.mockClear()
fn.mock.instances !== instances
fn.mock.instances.length === 0
instances.length === 1
```
Although perhaps this is the expected behavior